### PR TITLE
Tsk 3203 gen integrations readme

### PR DESCRIPTION
**Pull request by IdoGafi**

### CODE REVIEW

File: components/docs/markdown.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		}
+		buffer.WriteString("\n")
+
+	}
+	return buffer.String()
+}
+
+func NewTable(row, col int) *Table {
+	t := new(Table)
+	row = row + 2 // extra lines for title & line
```
Comment (blink-daniel) [inline components/docs/markdown.go:128]: row += 2

File: cmd/docs/main.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	}
+}
+
+func generate(path string) error {
+	return filepath.Walk(path, handleFile)
+}
+
+func handleFile(path string, f os.FileInfo, err error) error {
+	if err != nil {
+		return err
```
Comment (blink-daniel) [inline cmd/docs/main.go:31]: wrap the error with some context, e.g "handle file %s" with errors.Wrap

File: cmd/docs/main.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	} else if f.Name() == consts.DefaultMaskFileName {
+		// if it's a mask file
+		err = docs.GeneratePluginDocs(dir, consts.DefaultOpenapiFileName, consts.DefaultMaskFileName, "")
+	} else {
+		// if it's a gcp / azure mask
+		maskPrefix := strings.TrimSuffix(f.Name(), "-"+consts.DefaultMaskFileName)
+		err = docs.GeneratePluginDocs(dir, maskPrefix+".yaml", f.Name(), "")
+	}
+	if err != nil {
+		return err
```
Comment (blink-daniel) [inline cmd/docs/main.go:51]: wrap the error with some context, e.g "handle file %s" with errors.Wrap

File: components/docs/docs.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	descTitle    = "Description"
+	ParamTitle   = "Parameter"
+)
+
+func GeneratePluginDocs(pathToPlugin string, openapiPath string, maskPath string, namedActionsPath string) error {
+	fullPath := openapi.GetFullPath(pathToPlugin)
+
+	plugin, err := openapi.GetPlugin(fullPath, openapiPath, maskPath, namedActionsPath)
+	if err != nil {
+		return err
```
Comment (blink-daniel) [inline components/docs/docs.go:23]: same here, add context to the error

File: components/docs/docs.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+			t.SetContent(rowNum, paramColumn, maskedParam.DisplayName)
+			t.SetContent(rowNum, descColumn, maskedParam.Description)
+			rowNum++
+		}
+		mkdwn.WriteTable(t)
+	}
+
+	err = mkdwn.ExportToFile("./docs/" + plugin.CollectionName + ".md")
+	if err != nil {
+		log.Fatal(err)
```
Comment (blink-daniel) [inline components/docs/docs.go:47]: intentional instead of simply returning an error?

File: components/docs/markdown.go [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	m := new(MarkdownDoc)
+	m.doc = new(strings.Builder)
+	return m
+}
+
+func (m *MarkdownDoc) ExportToFile(filename string) error {
+	docToExport := m.doc.String()
+	// replacing characters according to docuzaurus limitations
+	docToExport = AdjustTextForDocusaurus(docToExport)
+	return ioutil.WriteFile(filename, []byte(docToExport), os.ModePerm)
```
Comment (blink-daniel) [inline components/docs/markdown.go:26]: add error handling so we'll know the context of the error, e.g

if err := ioutil.WriteFile(filename, []byte(docToExport), os.ModePerm); err != nil {
   return errors.Wrap(err, "export to file")
}

return nil

File: components/docs/markdown.go [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+const lineBreak = "<br/>\n"
+
+type MarkdownDoc struct {
+	doc *strings.Builder
+}
+
+func NewMarkDown() *MarkdownDoc {
+	m := new(MarkdownDoc)
+	m.doc = new(strings.Builder)
+	return m
```
[the remark this answers is not in the document]
Comment (blink-daniel) [reply components/docs/markdown.go:19]: FYI - the only time we have to use new() is for passing primtives pointers, such as new(int)

**Review by blink-daniel — CHANGES_REQUESTED** [review]
few minor changes, also regarding error handling make sure to follow https://github.com/blinkops/style/blob/main/style.md#error-wrapping

File: components/docs/docs.go (hunk 5 above)
Comment (IdoGafi) [reply components/docs/docs.go:47]: Yep, if something is not working write we would like not to generate the docs at all instead of creating a broken one

**Review by blink-daniel — APPROVED** [review]

**Pull request merged**